tests/template/test_load_context.py

```diff
@@ -51,14 +51,32 @@ def test_valid_context(self, fake_repo_path):
         assert str(fake_repo_path.resolve() / "src") in sys.path
 
     def test_valid_context_with_env(self, mocker, monkeypatch, fake_repo_path):
-        """Test getting project context when Kedro config environment is
-        specified in the environment variable.
-        """
+        """Test getting project context when Kedro config environment is specified
+        in the environment variable."""
+        # Disable logging.config.dictConfig in KedroContext._setup_logging as
+        # it changes logging.config and affects other unit tests
+        mocker.patch("logging.config.dictConfig")
         mocker.patch("kedro.config.config.ConfigLoader.get")
         monkeypatch.setenv("KEDRO_ENV", "my_fake_env")
         result = load_context(str(fake_repo_path))
         assert result.env == "my_fake_env"
 
+    def test_loading_valid_context_twice(self, mocker, fake_repo_path):
+        """Test getting project context twice."""
+        mocker.patch("logging.config.dictConfig")
+
+        # first time
+        load_context(str(fake_repo_path))
+
+        try:
+            # second time
+            load_context(str(fake_repo_path))
+        except KedroContextError as kce:  # pragma: no cover
```

Comment:
What is the exception raised when `load_context()` fails, other than `KedroContextError`?

Reply:
My point here was that I believe we don't need to catch the error here just to run `pytest.fail()`. I think we should let the error bubble up as pytest will catch it anyways. This will reduce the complexity of test code and number of `# pragma: no cover`s with the same result.

Reply:
What you are suggesting here although might appear to save a few lines of code and in your view reduce complexity, although in my view it opens up the scenarios where the test is not clear to read and understand (introduces assumptions) and also would flood the logs/console with stack traces (making less readable). The paradigm of _setup-action-assert_ and the likes are missing in the proposed format above. Also, by not catching the error(s) in the test we are asking the framework to handle it for us and this can go different ways.

I can expound on these points further if you like me to. Initially, Cheuk also introduced checking for the values loaded (which we should have retained), as I think it is a sensible course of action, to cover for silent failures. By which I mean, what if loading does not result in exceptions but invalid values are loaded and we let it through. While you may say that may not be the test we are working and our code is covering for it. In my view from past experience, doing a bit more in a test is always better than not. Also makes the intent of checking clearer for those who come after us. There will always be tests that marginally overall in intent with another bunch of them.

I can give you examples of outputs and other scenarios if you like me to but it would only make this PR longer than it should have been.

I'm happy to cover for the additional tests that have opened up with the branch flag being enabled - hopefully, we can move forward with this one. I propose the below:

```python
    def test_loading_valid_context_twice(self, mocker, fake_repo_path):
        """Test getting project context twice."""
        mocker.patch("logging.config.dictConfig")

        # first time
        try:
            # first time
            load_context(str(fake_repo_path))
            ---> (assert if the loaded context contains valid values, Cheuk's original work) <----
        except KedroContextError as kce:  # pragma: no cover
            pytest.fail(  # pragma: no cover
                "FAILED: Should have loaded the context the first-time successfully (1 out of 2 attempts), "
                "and not got this exception: " + str(kce)
            )
        load_context(str(fake_repo_path))

        try:
            # second time
            load_context(str(fake_repo_path))
            ---> (assert if the loaded context contains valid values, Cheuk's original work) <----
        except KedroContextError as kce:  # pragma: no cover
            pytest.fail(  # pragma: no cover
                "FAILED: Should have loaded the context twice successfully (2 out of 2 attempts), "
                "and not got this exception: " + str(kce)
            )
```
_(with of course applying refactorings of semantically repeated code above)_

I like `kedro`, and everyone else as well. I would not like to introduce gaps in it when I can see it in sight.